aws/auth_helpers.go

```diff
@@ -125,8 +126,19 @@ func GetCredentials(c *Config) (*awsCredentials.Credentials, error) {
 	// Build isolated HTTP client to avoid issues with globally-shared settings
 	client := cleanhttp.DefaultClient()
 
-	// Keep the timeout low as we don't want to wait in non-EC2 environments
-	client.Timeout = 100 * time.Millisecond
+	// Keep the default timeout low as we don't want to wait in non-EC2 environments
+	timeoutMillis := 100
+	timeoutString := os.Getenv("AWS_METADATA_TIMEOUT_MILLISECONDS")
```

Comment:
I would set the env variable name as a constant for reusability here and in the log